Code/GraphMol/MolStandardize/Normalize.cpp

```diff
@@ -25,24 +30,31 @@ class ROMol;
 
 namespace MolStandardize {
 
+typedef boost::flyweight<
+    boost::flyweights::key_value<std::string, TransformCatalogParams>,
+    boost::flyweights::no_tracking>
+    param_flyweight;
+
 // unsigned int MAX_RESTARTS = 200;
 
 // constructor
 Normalizer::Normalizer() {
   BOOST_LOG(rdInfoLog) << "Initializing Normalizer\n";
-  TransformCatalogParams tparams(defaultCleanupParameters.normalizations);
+  const TransformCatalogParams *tparams =
+      &(param_flyweight(defaultCleanupParameters.normalizations).get());
   //  unsigned int ntransforms = tparams->getNumTransformations();
   //  TEST_ASSERT(ntransforms == 22);
-  this->d_tcat = new TransformCatalog(&tparams);
+  this->d_tcat = new TransformCatalog(tparams);
   this->MAX_RESTARTS = 200;
 }
 
 // overloaded constructor
 Normalizer::Normalizer(const std::string normalizeFile,
                        const unsigned int maxRestarts) {
   BOOST_LOG(rdInfoLog) << "Initializing Normalizer\n";
-  TransformCatalogParams tparams(normalizeFile);
-  this->d_tcat = new TransformCatalog(&tparams);
+  const TransformCatalogParams *tparams =
```

Comment:
It strikes me that if this is a bottle neck, the filename is not the right place to optimize, but rather the TransformCatalog.

Can't one look up a const transformcatalog given a key (filename) value?

Reply:
That appears to be what you are doing, but I'm unclear about how a const tparams is initialized if it isn't already int the flyweight lookup.

Reply:
What's here is already a significant optimization. Moving it so that we do flyweights of the transform catalogs themselves keyed by the parameter file name would help more

Reply:
To make that concrete: The current code (on master) spent 1/3 of its run-time in `SmartsToMol`. This version has that down to <7%, so the low-hanging fruit is plucked.